src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  memo,
+  useEffect,
+  useState,
+  useCallback,
+  useMemo,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { getPeople } from '../../api';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable/PeopleTable';
+import { PeopleFilters } from '../PeopleFilters/PeopleFilters';
+import { Person } from '../../types';
+import { Sex } from '../../types/SexEnum';
+import { SortType } from '../../types/SortType';
+
+const findParent = (people: Person[], parentName: string | null) => {
+  return people.find((person) => person.name === parentName);
+};
+
+export const PeoplePage = memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [searchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex') || Sex.ALL;
+  const query = searchParams.get('query')?.toLowerCase() || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const order = searchParams.get('order');
+  const sort = searchParams.get('sort');
+
+  const loadPeople = useCallback(async () => {
```

Comment:
As you use this function only inside the useEffect you may move it there. Don't just use useCallback without a reason.

![image](https://github.com/mate-academy/react_people-table-advanced/assets/102293862/bb281729-07e6-4e2d-b756-3c9122324c2e)

https://react.dev/reference/react/useCallback

I'd recommend you read more about the useCallback hook and then fix all the cases when you use it without a reason.